tss-esapi/src/structures/buffers/public.rs

```diff
@@ -111,52 +112,40 @@ impl TryFrom<Public> for PublicBuffer {
 impl Marshall for PublicBuffer {
     const BUFFER_SIZE: usize = std::mem::size_of::<TPM2B_PUBLIC>();
 
-    /// Produce a marshalled [`TPM2B_PUBLIC`]
-    fn marshall(&self) -> Result<Vec<u8>> {
-        let mut buffer = vec![0; Self::BUFFER_SIZE];
-        let mut offset = 0;
-
+    fn marshall_offset(&self, marshalled_data: &mut [u8], offset: &mut usize) -> Result<()> {
```

Comment:
This one doesn't fit the macro patterns?

Reply:
Fixed